src/main.rs

```diff
@@ -32,65 +32,65 @@ const CARGO_PKG_NAME: &str = env!("CARGO_PKG_NAME");
 const CARGO_PKG_VERSION: &str = env!("CARGO_PKG_VERSION");
 
 #[derive(Debug, Parser)]
-#[clap(name = format!("{CARGO_PKG_NAME} (v{CARGO_PKG_VERSION})"))]
+#[clap(name = CARGO_PKG_NAME)]
 #[clap(about = "An extremely fast Python linter.", long_about = None)]
 #[clap(version)]
 struct Cli {
-    #[clap(parse(from_os_str), value_hint = ValueHint::AnyPath, required = true)]
+    #[arg(value_hint = ValueHint::AnyPath, required = true)]
     files: Vec<PathBuf>,
     /// Enable verbose logging.
-    #[clap(short, long, action)]
+    #[arg(short, long, action)]
     verbose: bool,
     /// Disable all logging (but still exit with status code "1" upon detecting errors).
-    #[clap(short, long, action)]
+    #[arg(short, long, action)]
     quiet: bool,
     /// Exit with status code "0", even upon detecting errors.
-    #[clap(short, long, action)]
+    #[arg(short, long, action)]
     exit_zero: bool,
     /// Run in watch mode by re-running whenever files change.
-    #[clap(short, long, action)]
+    #[arg(short, long, action)]
     watch: bool,
     /// Attempt to automatically fix lint errors.
-    #[clap(short, long, action)]
+    #[arg(short, long, action)]
     fix: bool,
     /// Disable cache reads.
-    #[clap(short, long, action)]
+    #[arg(short, long, action)]
     no_cache: bool,
     /// List of error codes to enable.
-    #[clap(long, multiple = true)]
+    #[arg(long, num_args = 1..)]
     select: Vec<CheckCode>,
```

Comment:
How come you accept both `--select code --select code` and `--select code code`?

Normally appls just accept `--select code,code --select code` instead.

Reply:
I'd like to only accept `--select code+`, is that not what's happening here?

Reply:
(As in: either `--select F401` or `--select F401 E501` should be valid .)

Reply:
Could you write that out further to make sure I understand?

Reply:
Yes, sorry. The intent is that either `--select F401` or `--select F401 E501` should be accepted. `--select F401 --select E501` shouldn't be though (and neither should `--select F401,E501`).

Though, regardless of whether that aligns with the _actual_ behavior here, perhaps that intent is misguided?

Reply:
I note that `flake8` accepts `--select F401,E501` and not `--select F401 E501` or `--select F401 --select E501`.

The `--select F401 E501` syntax confused me at first because it prevents you from writing `ruff --select F401 file.py`. Generally UNIX options only ever take a known number of arguments, and this is why.

Reply:
`--select F401 E501` without multiple occurrences would requiring overriding the default `ArgAction` for `Vec` (`Append`) with `action = ArgAction::Set`.

And yes, this general leads to ambiguities in parsing which is why we [place warnings around it](https://docs.rs/clap/latest/clap/builder/struct.Arg.html#method.num_args).

I would recommend instead `#[arg(long, value_delimiter = ",")]` or `#[arg(long, value_delimiter = ",", action = ArgAction::Set)]`, depending on if your want to allow multiple occurrences.

Reply:
Thank you both, much appreciated + I'm convinced. Changed in #278.